MacOS：Building CXX object src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsLogical.cpp.o  Fail

OS: mac OS 10.14.6
gcc verson : gcc (GCC) 10.0.1
ninja version: 1.10.0

From Docs i know, "Build should work on Mac OS X 10.15 (Catalina)." But I heard that there will be problems after upgrading MacOS to 15, so macOS has not been upgraded. I encountered the following problem, so I want to know if it is related to the OS version? And how to solve the problem?


**# ninja clickhouse-server clickhouse-client**
```
[0/2] Re-checking globbed directories...
[267/844] Building CXX object src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsHashing.cpp.o
FAILED: src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsHashing.cpp.o 
/usr/bin/clang++  -DENABLE_MULTITARGET_CODE=1 -DLZ4_DISABLE_DEPRECATE_WARNINGS=1 -DPOCO_ENABLE_CPP11 -DPOCO_HAVE_FD_POLL -DPOCO_OS_FAMILY_UNIX -DUNALIGNED_OK -DUSE_FASTMEMCPY=0 -DUSE_HYPERSCAN=1 -DUSE_REPLXX=1 -DUSE_XXHASH=1 -DWITH_COVERAGE=0 -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -Iincludes/configs -I../contrib/cityhash102/include -I../contrib/libfarmhash -I../src -Isrc -Isrc/Core/include -I../base/common/.. -Ibase/common/.. -I../contrib/cctz/include -Icontrib/zlib-ng -I../contrib/zlib-ng -I../base/pcg-random/. -I../contrib/consistent-hashing -I../contrib/consistent-hashing-sumbur -I../contrib/libuv/include -I../contrib/libmetrohash/src -I../contrib/murmurhash/include -I../contrib/lz4/lib -isystem ../contrib/sparsehash-c11 -isystem ../contrib/h3/src/h3lib/include -isystem ../contrib/rapidjson/include -isystem ../contrib/libcxx/include -isystem ../contrib/libcxxabi/include -isystem ../contrib/base64 -isystem ../contrib/msgpack-c/include -isystem ../contrib/re2 -isystem ../contrib/boost -isystem ../contrib/poco/Net/include -isystem ../contrib/poco/Foundation/include -isystem ../contrib/poco/NetSSL_OpenSSL/include -isystem ../contrib/poco/Crypto/include -isystem ../contrib/openssl-cmake/linux_x86_64/include -isystem ../contrib/openssl/include -isystem ../contrib/poco/Util/include -isystem ../contrib/poco/JSON/include -isystem ../contrib/poco/XML/include -isystem ../contrib/replxx/include -isystem ../contrib/fmtlib-cmake/../fmtlib/include -isystem ../contrib/double-conversion -isystem ../contrib/ryu -isystem contrib/re2_st -isystem ../contrib/croaring -isystem ../contrib/orc/c++/include -isystem contrib/orc/c++/include -isystem ../contrib/pdqsort -isystem ../contrib/AMQP-CPP/include -isystem ../contrib/libdivide/. -isystem contrib/h3/src/h3lib/include -isystem ../contrib/hyperscan/src -isystem ../contrib/simdjson/include -isystem ../contrib/stats/include -isystem ../contrib/gcem/include -fchar8_t -fdiagnostics-color=always -std=c++2a -fsized-deallocation  -gdwarf-aranges -msse4.1 -msse4.2 -mpopcnt   -Wall -Wno-unused-command-line-argument  -stdlib=libc++ -fdiagnostics-absolute-paths -Werror -mmacosx-version-min=10.15 -Wextra -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wcomma -Wconditional-uninitialized -Wcovered-switch-default -Wdeprecated -Wembedded-directive -Wextra-semi -Wgnu-case-range -Winconsistent-missing-destructor-override -Wnewline-eof -Wold-style-cast -Wrange-loop-analysis -Wredundant-parens -Wreserved-id-macro -Wshadow-field -Wshadow-uncaptured-local -Wshadow -Wstring-plus-int -Wundef -Wunreachable-code-return -Wunreachable-code -Wunused-exception-parameter -Wunused-macros -Wunused-member-function -Wzero-as-null-pointer-constant -Weverything -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-return-std-move-in-c++11 -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -g -O0 -g3 -ggdb3 -fno-inline  -D_LIBCPP_DEBUG=0 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk   -D OS_DARWIN -nostdinc++ -MD -MT src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsHashing.cpp.o -MF src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsHashing.cpp.o.d -o src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsHashing.cpp.o -c ../src/Functions/FunctionsHashing.cpp
In file included from ../src/Functions/FunctionsHashing.cpp:1:
In file included from ../src/Functions/FunctionsHashing.h:44:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:197:1: error: expected unqualified-id
DECLARE_SSE42_SPECIFIC_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:131:42: note: expanded from macro 'DECLARE_SSE42_SPECIFIC_CODE'
#define DECLARE_SSE42_SPECIFIC_CODE(...) \
                                         ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:102:9: note: expanded from macro '\
BEGIN_SSE42_SPECIFIC_CODE'
        _Pragma("clang attribute push(__attribute__((target(\"sse,sse2,sse3,ssse3,sse4,popcnt\"))),apply_to=function)")
        ^
<scratch space>:53:8: note: expanded from here
 clang attribute push(__attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt"))),apply_to=function)
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
In file included from ../src/Functions/FunctionsHashing.h:44:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:197:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:137:3: note: expanded from macro 'DECLARE_SSE42_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:55:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
In file included from ../src/Functions/FunctionsHashing.h:44:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:201:1: error: expected unqualified-id
DECLARE_AVX_SPECIFIC_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:146:3: note: expanded from macro 'DECLARE_AVX_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:59:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
In file included from ../src/Functions/FunctionsHashing.h:44:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:205:1: error: expected unqualified-id
DECLARE_AVX2_SPECIFIC_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:155:3: note: expanded from macro 'DECLARE_AVX2_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:63:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
In file included from ../src/Functions/FunctionsHashing.h:44:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:209:1: error: expected unqualified-id
DECLARE_AVX512F_SPECIFIC_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:164:3: note: expanded from macro 'DECLARE_AVX512F_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:67:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:595:1: error: expected unqualified-id
DECLARE_MULTITARGET_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:187:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_DEFAULT_CODE         (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:131:42: note: expanded from macro '\
DECLARE_SSE42_SPECIFIC_CODE'
#define DECLARE_SSE42_SPECIFIC_CODE(...) \
                                         ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:102:9: note: expanded from macro '\
BEGIN_SSE42_SPECIFIC_CODE'
        _Pragma("clang attribute push(__attribute__((target(\"sse,sse2,sse3,ssse3,sse4,popcnt\"))),apply_to=function)")
        ^
<scratch space>:259:8: note: expanded from here
 clang attribute push(__attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt"))),apply_to=function)
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:595:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:187:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_DEFAULT_CODE         (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:137:3: note: expanded from macro '\
DECLARE_SSE42_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:261:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:595:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:188:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_SSE42_SPECIFIC_CODE  (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:146:3: note: expanded from macro '\
DECLARE_AVX_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:265:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:595:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:189:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_AVX_SPECIFIC_CODE    (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:155:3: note: expanded from macro '\
DECLARE_AVX2_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:269:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:595:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:190:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_AVX2_SPECIFIC_CODE   (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:164:3: note: expanded from macro '\
DECLARE_AVX512F_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:273:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:706:1: error: expected unqualified-id
DECLARE_MULTITARGET_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:187:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_DEFAULT_CODE         (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:131:42: note: expanded from macro '\
DECLARE_SSE42_SPECIFIC_CODE'
#define DECLARE_SSE42_SPECIFIC_CODE(...) \
                                         ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:102:9: note: expanded from macro '\
BEGIN_SSE42_SPECIFIC_CODE'
        _Pragma("clang attribute push(__attribute__((target(\"sse,sse2,sse3,ssse3,sse4,popcnt\"))),apply_to=function)")
        ^
<scratch space>:278:8: note: expanded from here
 clang attribute push(__attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt"))),apply_to=function)
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:706:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:187:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_DEFAULT_CODE         (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:137:3: note: expanded from macro '\
DECLARE_SSE42_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:280:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:706:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:188:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_SSE42_SPECIFIC_CODE  (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:146:3: note: expanded from macro '\
DECLARE_AVX_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:4:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:706:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:189:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_AVX_SPECIFIC_CODE    (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:155:3: note: expanded from macro '\
DECLARE_AVX2_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:8:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:706:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:190:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_AVX2_SPECIFIC_CODE   (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:164:3: note: expanded from macro '\
DECLARE_AVX512F_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:12:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsHashing.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:1297:25: error: no template named 'FunctionAnyHash'; did you mean 'TargetSpecific::Default::FunctionAnyHash'?
using FunctionHalfMD5 = FunctionAnyHash<HalfMD5Impl>;
                        ^~~~~~~~~~~~~~~
                        TargetSpecific::Default::FunctionAnyHash
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:709:7: note: 'TargetSpecific::Default::FunctionAnyHash' declared here
class FunctionAnyHash : public IFunction
      ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:1299:27: error: no template named 'FunctionAnyHash'; did you mean 'TargetSpecific::Default::FunctionAnyHash'?
using FunctionSipHash64 = FunctionAnyHash<SipHash64Impl>;
                          ^~~~~~~~~~~~~~~
                          TargetSpecific::Default::FunctionAnyHash
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:709:7: note: 'TargetSpecific::Default::FunctionAnyHash' declared here
class FunctionAnyHash : public IFunction
      ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:1300:27: error: no template named 'FunctionIntHash'; did you mean 'TargetSpecific::Default::FunctionIntHash'?
using FunctionIntHash32 = FunctionIntHash<IntHash32Impl, NameIntHash32>;
                          ^~~~~~~~~~~~~~~
                          TargetSpecific::Default::FunctionIntHash
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:598:7: note: 'TargetSpecific::Default::FunctionIntHash' declared here
class FunctionIntHash : public IFunction
      ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:1301:27: error: no template named 'FunctionIntHash'; did you mean 'TargetSpecific::Default::FunctionIntHash'?
using FunctionIntHash64 = FunctionIntHash<IntHash64Impl, NameIntHash64>;
                          ^~~~~~~~~~~~~~~
                          TargetSpecific::Default::FunctionIntHash
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsHashing.h:598:7: note: 'TargetSpecific::Default::FunctionIntHash' declared here
class FunctionIntHash : public IFunction
      ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
[268/844] Building CXX object src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsRandom.cpp.o
FAILED: src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsRandom.cpp.o 
/usr/bin/clang++  -DENABLE_MULTITARGET_CODE=1 -DLZ4_DISABLE_DEPRECATE_WARNINGS=1 -DPOCO_ENABLE_CPP11 -DPOCO_HAVE_FD_POLL -DPOCO_OS_FAMILY_UNIX -DUNALIGNED_OK -DUSE_FASTMEMCPY=0 -DUSE_HYPERSCAN=1 -DUSE_REPLXX=1 -DUSE_XXHASH=1 -DWITH_COVERAGE=0 -DWITH_GZFILEOP -DX86_64 -DZLIB_COMPAT -Iincludes/configs -I../contrib/cityhash102/include -I../contrib/libfarmhash -I../src -Isrc -Isrc/Core/include -I../base/common/.. -Ibase/common/.. -I../contrib/cctz/include -Icontrib/zlib-ng -I../contrib/zlib-ng -I../base/pcg-random/. -I../contrib/consistent-hashing -I../contrib/consistent-hashing-sumbur -I../contrib/libuv/include -I../contrib/libmetrohash/src -I../contrib/murmurhash/include -I../contrib/lz4/lib -isystem ../contrib/sparsehash-c11 -isystem ../contrib/h3/src/h3lib/include -isystem ../contrib/rapidjson/include -isystem ../contrib/libcxx/include -isystem ../contrib/libcxxabi/include -isystem ../contrib/base64 -isystem ../contrib/msgpack-c/include -isystem ../contrib/re2 -isystem ../contrib/boost -isystem ../contrib/poco/Net/include -isystem ../contrib/poco/Foundation/include -isystem ../contrib/poco/NetSSL_OpenSSL/include -isystem ../contrib/poco/Crypto/include -isystem ../contrib/openssl-cmake/linux_x86_64/include -isystem ../contrib/openssl/include -isystem ../contrib/poco/Util/include -isystem ../contrib/poco/JSON/include -isystem ../contrib/poco/XML/include -isystem ../contrib/replxx/include -isystem ../contrib/fmtlib-cmake/../fmtlib/include -isystem ../contrib/double-conversion -isystem ../contrib/ryu -isystem contrib/re2_st -isystem ../contrib/croaring -isystem ../contrib/orc/c++/include -isystem contrib/orc/c++/include -isystem ../contrib/pdqsort -isystem ../contrib/AMQP-CPP/include -isystem ../contrib/libdivide/. -isystem contrib/h3/src/h3lib/include -isystem ../contrib/hyperscan/src -isystem ../contrib/simdjson/include -isystem ../contrib/stats/include -isystem ../contrib/gcem/include -fchar8_t -fdiagnostics-color=always -std=c++2a -fsized-deallocation  -gdwarf-aranges -msse4.1 -msse4.2 -mpopcnt   -Wall -Wno-unused-command-line-argument  -stdlib=libc++ -fdiagnostics-absolute-paths -Werror -mmacosx-version-min=10.15 -Wextra -Wpedantic -Wno-vla-extension -Wno-zero-length-array -Wcomma -Wconditional-uninitialized -Wcovered-switch-default -Wdeprecated -Wembedded-directive -Wextra-semi -Wgnu-case-range -Winconsistent-missing-destructor-override -Wnewline-eof -Wold-style-cast -Wrange-loop-analysis -Wredundant-parens -Wreserved-id-macro -Wshadow-field -Wshadow-uncaptured-local -Wshadow -Wstring-plus-int -Wundef -Wunreachable-code-return -Wunreachable-code -Wunused-exception-parameter -Wunused-macros -Wunused-member-function -Wzero-as-null-pointer-constant -Weverything -Wno-c++98-compat-pedantic -Wno-c++98-compat -Wno-c99-extensions -Wno-conversion -Wno-deprecated-dynamic-exception-spec -Wno-disabled-macro-expansion -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-nested-anon-types -Wno-packed -Wno-padded -Wno-return-std-move-in-c++11 -Wno-shift-sign-overflow -Wno-sign-conversion -Wno-switch-enum -Wno-undefined-func-template -Wno-unused-template -Wno-vla -Wno-weak-template-vtables -Wno-weak-vtables -g -O0 -g3 -ggdb3 -fno-inline  -D_LIBCPP_DEBUG=0 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk   -D OS_DARWIN -nostdinc++ -MD -MT src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsRandom.cpp.o -MF src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsRandom.cpp.o.d -o src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsRandom.cpp.o -c ../src/Functions/FunctionsRandom.cpp
In file included from ../src/Functions/FunctionsRandom.cpp:1:
In file included from ../src/Functions/FunctionsRandom.h:6:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:197:1: error: expected unqualified-id
DECLARE_SSE42_SPECIFIC_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:131:42: note: expanded from macro 'DECLARE_SSE42_SPECIFIC_CODE'
#define DECLARE_SSE42_SPECIFIC_CODE(...) \
                                         ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:102:9: note: expanded from macro '\
BEGIN_SSE42_SPECIFIC_CODE'
        _Pragma("clang attribute push(__attribute__((target(\"sse,sse2,sse3,ssse3,sse4,popcnt\"))),apply_to=function)")
        ^
<scratch space>:179:8: note: expanded from here
 clang attribute push(__attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt"))),apply_to=function)
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
In file included from ../src/Functions/FunctionsRandom.h:6:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:197:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:137:3: note: expanded from macro 'DECLARE_SSE42_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:181:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
In file included from ../src/Functions/FunctionsRandom.h:6:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:201:1: error: expected unqualified-id
DECLARE_AVX_SPECIFIC_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:146:3: note: expanded from macro 'DECLARE_AVX_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:185:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
In file included from ../src/Functions/FunctionsRandom.h:6:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:205:1: error: expected unqualified-id
DECLARE_AVX2_SPECIFIC_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:155:3: note: expanded from macro 'DECLARE_AVX2_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:189:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
In file included from ../src/Functions/FunctionsRandom.h:6:
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:209:1: error: expected unqualified-id
DECLARE_AVX512F_SPECIFIC_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:164:3: note: expanded from macro 'DECLARE_AVX512F_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:193:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:39:1: error: expected unqualified-id
DECLARE_MULTITARGET_CODE(
^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:187:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_DEFAULT_CODE         (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:131:42: note: expanded from macro '\
DECLARE_SSE42_SPECIFIC_CODE'
#define DECLARE_SSE42_SPECIFIC_CODE(...) \
                                         ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:102:9: note: expanded from macro '\
BEGIN_SSE42_SPECIFIC_CODE'
        _Pragma("clang attribute push(__attribute__((target(\"sse,sse2,sse3,ssse3,sse4,popcnt\"))),apply_to=function)")
        ^
<scratch space>:96:8: note: expanded from here
 clang attribute push(__attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt"))),apply_to=function)
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:39:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:187:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_DEFAULT_CODE         (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:137:3: note: expanded from macro '\
DECLARE_SSE42_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:98:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:39:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:188:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_SSE42_SPECIFIC_CODE  (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:146:3: note: expanded from macro '\
DECLARE_AVX_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:102:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:39:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:189:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_AVX_SPECIFIC_CODE    (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:155:3: note: expanded from macro '\
DECLARE_AVX2_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:106:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:39:1: error: expected unqualified-id
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:190:44: note: expanded from macro 'DECLARE_MULTITARGET_CODE'
DECLARE_AVX2_SPECIFIC_CODE   (__VA_ARGS__) \
                                           ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:164:3: note: expanded from macro '\
DECLARE_AVX512F_SPECIFIC_CODE'
} \
  ^
/Users/CLionProjects/ClickHouse/src/Functions/TargetSpecific.h:104:9: note: expanded from macro '\
END_TARGET_SPECIFIC_CODE'
        _Pragma("clang attribute pop")
        ^
<scratch space>:110:8: note: expanded from here
 clang attribute pop
       ^
In file included from ../src/Functions/FunctionsRandom.cpp:1:
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:91:31: error: no template named 'FunctionRandomImpl'; did you mean 'FunctionRandom'?
class FunctionRandom : public FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>
                              ^~~~~~~~~~~~~~~~~~
                              FunctionRandom
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:91:7: note: 'FunctionRandom' declared here
class FunctionRandom : public FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>
      ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:91:31: error: too many template arguments for class template 'FunctionRandom'
class FunctionRandom : public FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>
                              ^                                                             ~~~~~
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:91:7: note: template is declared here
class FunctionRandom : public FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>
      ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:105:116: error: only virtual member functions can be marked 'override'
    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result, size_t input_rows_count) const override
                                                                                                                   ^~~~~~~~
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:97:13: error: no template named 'FunctionRandomImpl'; did you mean '::DB::FunctionRandom'?
            FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>>();
            ^~~~~~~~~~~~~~~~~~
            ::DB::FunctionRandom
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:91:7: note: '::DB::FunctionRandom' declared here
class FunctionRandom : public FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>
      ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:97:13: error: too many template arguments for class template 'FunctionRandom'
            FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>>();
            ^                                                             ~~~~~
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:91:7: note: template is declared here
class FunctionRandom : public FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>
      ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:97:81: error: expected unqualified-id
            FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>>();
                                                                                ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:101:13: error: no template named 'FunctionRandomImpl'; did you mean '::DB::FunctionRandom'?
            FunctionRandomImpl<TargetSpecific::AVX2::RandImpl, ToType, Name>>();
            ^~~~~~~~~~~~~~~~~~
            ::DB::FunctionRandom
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:91:7: note: '::DB::FunctionRandom' declared here
class FunctionRandom : public FunctionRandomImpl<TargetSpecific::Default::RandImpl, ToType, Name>
      ^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:101:48: error: no member named 'AVX2' in namespace 'DB::TargetSpecific'
            FunctionRandomImpl<TargetSpecific::AVX2::RandImpl, ToType, Name>>();
                               ~~~~~~~~~~~~~~~~^
/Users/CLionProjects/ClickHouse/src/Functions/FunctionsRandom.h:101:80: error: expected a type
            FunctionRandomImpl<TargetSpecific::AVX2::RandImpl, ToType, Name>>();
                                                                               ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
[272/844] Building CXX object src/Functions/CMakeFiles/clickhouse_functions.dir/FunctionsLogical.cpp.o
ninja: build stopped: subcommand failed.
```

--- a/cmake/tools.cmake
+++ b/cmake/tools.cmake
@@ -15,6 +15,10 @@ if (COMPILER_GCC)
 elseif (COMPILER_CLANG)
     # Require minimum version of clang/apple-clang
     if (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
+        # If you are developer you can figure out what exact versions of AppleClang are Ok,
+        # remove the following line and commit changes below.
+        message (FATAL_ERROR "AppleClang is not supported, you should install clang from brew.")
+
         # AppleClang 10.0.1 (Xcode 10.2) corresponds to LLVM/Clang upstream version 7.0.0
         # AppleClang 11.0.0 (Xcode 11.0) corresponds to LLVM/Clang upstream version 8.0.0
         set (XCODE_MINIMUM_VERSION 10.2)

